File: KCD/HMIgnoreZeroTransformer.m
//
//  HMIgnoreZeroTransformer.m
//  KCD
//
//  Created by Hori,Masaki on 2014/10/25.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMIgnoreZeroTransformer.h"

@implementation HMIgnoreZeroTransformer
+ (void)load
{
	static dispatch_once_t onceToken;
	dispatch_once(&onceToken, ^{
		[NSValueTransformer setValueTransformer:[self new] forName:@"HMIgnoreZeroTransformer"];
	});
}
+ (Class)transformedValueClass
{
	return [NSNumber class];
}
+ (BOOL)allowsReverseTransformation
{
	return NO;
}

- (id)transformedValue:(id)value
{
	if(![value isKindOfClass:[NSNumber class]]) return nil;
	
	if([value integerValue] == 0) return nil;
	
	return value;
}
@end

File: KCD/HMIgnoreZeroTransformer.swift
//
//  HMIgnoreZeroTransformer.swift
//  KCD
//
//  Created by Hori,Masaki on 2014/12/21.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMIgnoreZeroTransformer: NSValueTransformer
{
	override class func load() {
		NSValueTransformer.setValueTransformer(HMIgnoreZeroTransformer(), forName: "HMIgnoreZeroTransformer")
	}
	
	override class func transformedValueClass() -> AnyClass {
		return NSNumber.self
	}
	
	override class func allowsReverseTransformation() -> Bool { return false }
	
	override func transformedValue(value: AnyObject?) -> AnyObject? {
		let numValue = value as? NSNumber
		if numValue == nil {
			return nil
		}
		if numValue == 0 {
			return nil
		}
		return value
	}
}
